MeetupPlatformApi/Controllers/AuthenticationController.cs

```diff
@@ -30,13 +30,21 @@ public async Task<IActionResult> GetUserById(Guid id)
     {
         var user = await context.Users.SingleOrDefaultAsync(user => user.Id == id);
         var outputDto = mapper.Map<UserOutputDto>(user);
-        return outputDto is null ? Ok(outputDto) : NotFound();
+        return outputDto is not null ? Ok(outputDto) : NotFound();
     }
 
     [HttpPost]
-    public async Task<IActionResult> RegisterUser([FromBody] UserRegistrationDto userFromBody)
+    public async Task<IActionResult> RegisterUser([FromBody] UserRegistrationDto registrationDto)
     {
-        var user = mapper.Map<UserEntity>(userFromBody);
+        var usernameAlreadyTaken = await context.Users.AnyAsync(user => user.Username == user.Username);
```

Comment:
You're comparing identical object instances.
Change one of them from user.Username to registrationDto.Username